packages/flutter/lib/src/material/tab_controller.dart

```diff
@@ -205,6 +205,18 @@ class TabController extends ChangeNotifier {
   /// [TabBarView.children]'s length.
   final int length;
 
+  /// The duration given to [animateTo].
+  ///
+  /// Only valid while animation is running, null otherwise.
+  Duration get ongoingAnimationDuration => _ongoingAnimationDuration;
```

Comment:
Instead of exposing this, which may only be useful when the animation first started (if the animation is already on the flight for a while, knowing the duration alone is not enough)

I am thinking maybe the tabcontroller should expose a new api to register a new callback, something like `onAnimationStarted`. The callback will take in the duration, curve and index. In the tabview, it will register a callback to also start animation on _pageController

Reply:
Surely considerable. I was also not very fond of exposing ```ongoingAnimationDuration``` and ```ongoingAnimationCurve``` just for this purpose :( I'll try that out and see how it looks.
Perhaps I may also add ```onAnimationEnd``` along with ```onAnimationStart```?